cmd/gaia/app/hooks.go

```diff
@@ -0,0 +1,62 @@
+package app
```

Comment:
I feel as though we have moved this code before. Is there a good reason why this needs to be in the app, and not in the staking module?

Reply:
There might be a way...!

Reply:
done. great idea